### PR TITLE
support kube-proxy nftables mode

### PR DESCRIPTION
**Pull request by aojea**
/hold

wait until 1.31 graduates it to beta

Fixes: #3434

### CODE REVIEW

**Comment by aojea** [comment]
@BenTheElder the problem with this option is to keep compatibility with old version, as it only will work with new versions ... I can't remember if we have some precedent like this

**Comment by BenTheElder** [comment]
> @BenTheElder the problem with this option is to keep compatibility with old version, as it only will work with new versions ... I can't remember if we have some precedent like this

This is true with feature gates, not exactly the same as those are totally user supplied strings.

File: site/content/docs/user/configuration.md [hunk 1]
```diff
@@ -217,14 +217,14 @@ networking:
 
 #### kube-proxy mode
 
-You can configure the kube-proxy mode that will be used, between iptables and ipvs. By
-default iptables is used
+You can configure the kube-proxy mode that will be used, between iptables, ipvs and nftables.
```
Comment (BenTheElder) [inline site/content/docs/user/configuration.md:220]:
```suggestion
You can configure the kube-proxy mode that will be used, between iptables, nftables (Kubernetes v1.30+), and ipvs.
```

**Review by BenTheElder — COMMENTED** [review]
Actually, we will need to internally version-check at cluster config time anyhow and enable NFTablesProxyMode feature gate? Or do we expect the user to do this?

It's new that we have our own config option for an alpha feature.

**Comment by danwinship** [comment]
> It's new that we have our own config option for an alpha feature.

no, you missed:

> wait until 1.31 graduates it to beta

**Comment by BenTheElder** [comment]
Sorry, yes, SGTM.

I think we should just document that it's 1.31+, we allow clusters with mixed versions and only advanced users will be setting this field anyhow.

**Comment by aojea** [comment]
/hold cancel

K/K PR is already merged and this will allow to set up CI without having to use additional scripting https://github.com/kubernetes/kubernetes/pull/124383

**Pull request merged**